Tests/Common/Background Queue/QueueStorageTest.swift

```diff
@@ -92,17 +76,52 @@ class QueueStorageTest: UnitTest {
             blockingGroups: [.identifiedProfile(identifier: String.random)]
         )
 
-        XCTAssertEqual(fileStorageMock.saveCallsCount, 1) // only create task call
+        XCTAssertEqual(fileStorageMock.saveCallsCount, 2)
         XCTAssertFalse(actual.success)
         XCTAssertEqual(actual.queueStatus, QueueStatus(queueId: testSiteId, numTasksInQueue: 0))
     }
+}
 
-    func test_create_givenFileStorageDoesNotUpdateInventory_expectFalse() {
-        var returnValues = [true, false]
-        fileStorageMock.saveClosure = { _, _, _ in
-            returnValues.removeFirst()
-        }
+// MARK: integration tests
+
+#if !os(Linux) // LINUX_DISABLE_FILEMANAGER
+class QueueStorageIntegrationTest: UnitTest {
```

Comment:
Moved test functions from unit tests above into integration tests. 

I was having some errors running these test functions since I didn't mock enough. After reading the test functions, I felt the tests were higher quality by being integration tests using a real file system.